model/publishing/delivery/PublishingDeliveryService.php

```diff
@@ -79,7 +79,7 @@ public function publishDelivery(\core_kernel_classes_Resource $delivery)
                     new DeployTestEnvironments(),
```

Comment:
This task was called `DeployTestEnvironments` and I didn't rename it. Agree with you that the name does not represent any more what it does. Do you think we should rename it?
Regarding service - it looks like it's not used any more.

Reply:
Hmm. i would go for 
`PublicationTask` or `PublishingTask` or prefixing those with `Remote%`

If the service is not used, it's better to remove it.

What do you think?

Reply:
Created a new task and deprecated old one. I also removed unused methods and task to synchronize delivery properties. But I cannot remove `PublishingDeliveryService` because it's constants are still used. 

I have another branch where I'm trying to remove all this services, methods and configurations which are not used any more with new approach to publish delivery to remote environment, but it's not directly related to this PR and I didn't want to make it too big.